packages/compass/src/main/auto-update-manager.ts

```diff
@@ -1,12 +1,376 @@
-import AutoUpdateManager from 'hadron-auto-update-manager';
-import { ipcMain } from 'hadron-ipc';
+import { EventEmitter } from 'events';
+import os from 'os';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
 import COMPASS_ICON from './icon';
+import type { FeedURLOptions } from 'electron';
+import { app, dialog, BrowserWindow } from 'electron';
+import { setTimeout as wait } from 'timers/promises';
+import autoUpdater from './auto-updater';
+import preferences from 'compass-preferences-model';
+import got from 'got';
+import dl from 'electron-dl';
+import type { CompassApplication } from './application';
 
-import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
-const { log, mongoLogId, debug } = createLoggerAndTelemetry(
+const { log, mongoLogId, debug, track } = createLoggerAndTelemetry(
   'COMPASS-AUTO-UPDATES'
 );
 
+function getSystemArch() {
+  return process.platform === 'darwin'
+    ? os.cpus().some((cpu) => {
+        // process.arch / os.arch() will return the arch for which the node
+        // binary was compiled. Checking if one of the CPUs has Apple in its
+        // name is the way to check (there is slight difference between the
+        // earliest models naming and a current one, so we check only for
+        // Apple in the name)
+        return /Apple/.test(cpu.model);
+      })
+      ? 'arm64'
+      : 'x64'
+    : process.arch;
+}
+
+async function promptForUpdate(
+  from: string,
+  to: string
+): Promise<'download' | 'update' | 'cancel'> {
+  const isMismatchedArchDarwin =
+    process.platform === 'darwin' && getSystemArch() !== process.arch;
+  const commonOptions = {
+    icon: COMPASS_ICON,
+    title: 'New version available',
+    message: 'A new version of Compass is available to install',
+  };
+
+  if (!isMismatchedArchDarwin) {
+    const answer = await dialog.showMessageBox({
+      ...commonOptions,
+      detail: `Compass ${to} is available. You are currently using ${from}. Would you like to download and install it now?`,
+      buttons: ['Install', 'Ask me later'],
+      cancelId: 1,
+    });
+
+    return answer.response === 0 ? 'update' : 'cancel';
+  }
+
+  const answer = await dialog.showMessageBox({
+    ...commonOptions,
+    detail: `Compass ${to} is available. You are currently using a build of Compass that is not optimized for M1/M2 processors. Would you like to download the optimized arm64 version of Compass ${to} now?`,
```

Comment:
Do we also want to use `M1/M2` in the second sentence? I’m not sure how large the percentage of users is that doesn’t know that M1/M2 ~ arm64, but it might be worth using consistent language here

Reply:
Yeah, makes sense, I'll update it